literature_search.py

```diff
@@ -0,0 +1,243 @@
+                                       papers_df['abstract'].astype(str)+\
+                                       papers_df['keywords'].astype(str))
+
+titles_vec = vect.transform(papers_df['everything'])
+#OR if you don't want to use just the title:
+# titles_vec = vect.transform(papers_df['title'])
+
+#==============================================================================
+#========================= Predict topics =====================================
+# Predict Topics For Each Paper
+prediction_vec = model.predict(titles_vec)
+
+topics = []
+pred_val = []
+title_temp = []
+indx = []
+
+for k, top_val in enumerate(prediction_vec):
+    pred_val = np.max(top_val)
+    if pred_val > 0*np.sort(top_val)[-2]:
+        indx.append(k)
+        topics.append(le.inverse_transform([np.argmax(top_val)])[0])
+        title_temp.append(papers_df['title'][k])
+    else:
+        indx.append(k)
+        topics.append('unknown')
+        title_temp.append(papers_df['title'][k])
+papers_df = pd.DataFrame(data = {'title': title_temp,
+                                  'topic': topics})
+pd.set_option('display.max_colwidth', -1) #don't cut cell short in dataframe
+papers_df[['title','topic']].sample(5)
+pd.reset_option('display.max_colwidth')
+
+#==============================================================================
+#========================= Save Titles and Topics =============================
+
+#add info for github markdown format
+papers_df['title']   = [title if title[1] is not '[' else title[1:-1] for title in papers_df['title']]
+papers_df['authors'] = [authors[k] if authors[k][1] is not '[' else authors[1:-1] for k in indx]
+papers_df['journal'] = [journals[k] for k in indx]
+papers_df['links']   = [links[k]    for k in indx]
+#generate filename
+import datetime
+now = datetime.datetime.now()
+strings = [str(now.year), str(now.month), str(now.day),'litupdate.csv']
+fname = 'Literature_Updates/'+'-'.join(strings)
+strings = [str(now.year), str(now.month), str(now.day),'litupdate.md']
+mdname = 'Literature_Updates/'+'-'.join(strings)
+strings = [str(now.year), str(now.month), str(now.day),'litupdate']
+urlname = '-'.join(strings)
+
+print('Filename: ',fname)
+
+papers_df.head()
+papers_df.sort_values('topic').to_csv(fname, index = False)
+print('\nLiterature Update Exported as .csv')
+
+# Compile papers grouped by topic
+md_file = open(mdname, 'w', encoding = 'utf-8')
+md_file.write('---\n')
+md_file.write('title: Biomechanics Literature Update\n')
+md_file.write('collection: literature\n')
+md_file.write('permalink: /literature/%s\n' % urlname)
+md_file.write('excerpt: <br>\n')
+md_file.write('toc: true\n')
+md_file.write('toc_sticky: true\n')
+md_file.write('toc_label: Topics\n')
+md_file.write('---\n')
+
+#tidy up topic strings
+topic_list = np.unique(papers_df.sort_values('topic')['topic'])
+ss = [s for s in topic_list if 'UNIQUE' in s]
+for i,t in enumerate(topic_list):
+    if 'UNIQUE' in t:  
```

Comment:
we're including unique topic now?

Reply:
Thats how it is in the ipynb, I didn't change that. 

I believe this should never actually be called because there are not UNIQUE in the topic list, so that if statement will never be true.

Reply:
then let's just cut it.

Reply:
I think leave it, cause we may one day try to implement the unique topic.